File: model/debugbar.go
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

package model

import "runtime"

type DebugBarInfo struct {
	ServerOS             string
	ServerArchitecture   string
	ServerVersion        string
	BuildHash            string
	DatabaseType         string
	DatabaseVersion      string
	LdapVendorName       string
	LdapVendorVersion    string
	ElasticServerVersion string
	ElasticServerPlugins []string
	WebSocketConnections int
	MasterDBConnections  int
	ReadDBConnections    int
	SessionsCount        int64
	Goroutines           int
	Cpus                 int
	CgoCalls             int64
	GoVersion            string
	GoMemStats           runtime.MemStats
}

File: webapp/channels/src/packages/mattermost-redux/src/actions/debugbar.ts
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {ActionFunc, DispatchFunc} from 'mattermost-redux/types/actions';
import {DebugBarTypes} from 'mattermost-redux/action_types';

export function addLine(data: any): ActionFunc {
    return async (dispatch: DispatchFunc) => {
        return dispatch({
            data,
            type: DebugBarTypes.ADD_LINE,
        });
    };
}

export function clearLines(key?: string): ActionFunc {
    return async (dispatch: DispatchFunc) => {
        return dispatch({
            type: DebugBarTypes.CLEAR_LINES,
            key,
        });
    };
}
